lib/reek/smells/attribute.rb

```diff
@@ -32,21 +28,19 @@ def self.contexts # :nodoc:
         [:class, :module]
       end
 
-      def self.default_config
-        super.merge(SmellConfiguration::ENABLED_KEY => false)
-      end
-
       #
       # Checks whether the given class declares any attributes.
       #
       # @return [Array<SmellWarning>]
       #
       def examine_context(ctx)
+        @visiblity_tracker = {}
```

Comment:
1: Yes. It is ~~fixed~~ completely rewritten in #614.
2: We can't leave it in the constructor because visibility tracking needs te be reset for every context.